pkg/controller/registry/resolver/cache/predicates.go

```diff
@@ -282,6 +281,32 @@ func (p orPredicate) String() string {
 	return b.String()
 }
 
+func Not(p ...Predicate) Predicate {
```

Comment:
this is a bit confusing to me. `Not` is usually an operator on a predicate not a list of predicates. Looking at the `Test()` it seems to imply that the list of predicate is implied to be a conjunction. Lmk if I'm missing something but coming into this cold I would have expected to use `Not` on a single predicate. If not, could we name it `None` or `Neither` or something.

Reply:
I think the same logic (no pun intended) applies to "and" vs. "all", and "or" vs. "any", but that's how these resolver primitives were named. I'm just following the current naming scheme.